### ISSUE TITLE
[MU4 Issue] Shortcuts aren't mapped to correct functions on macOS

### ISSUE THREAD

**Issue by David-C0pperfield** [musescore/MuseScore#15033]
In MuseScore 4 RC, some shortcuts are not mapped to the right place.

# Reproduction

1. Reset to factory settings.
2. Create a new score / Open an existed score.
3. Use certain shortcuts, which are listed below.
4. See the table issue

| Shortcut | Behavior | Behavior in MS4 RC | Actual key to have the right behavior in MS4 |
| ------ | ------ | ------ | ------ |
| "{"and "}" | in-/de-crease the measure length | Nothing | "[" and "]"|
| A | Input note A | Back to the beginning of current measure | None |

* This issue will be updated if more details are found

 - OS: macOS 13.1

**Comment by erinic04** [comment]
t is the key to add ties now, + does sharps, = does naturals, and - does flats.

**Comment by MarcSabatella** [comment]
Cmd+I is not a supported command anymore.  It wasn't removed from the code yet, but it will be.  Or it will be changed to a command that works :-).

As mentioned, +-/= are functioning correctly - the behavior you observe is by design.

But the behavior with "A" and with { or } is not a known issue.  The shortcut for reduce stretch is supposed to be "{".  On most keyboards, this is the same as Shift+[.  Is that not how it works on your keyboard?  Please explain more about the keyboard layout you are using.

**Comment by David-C0pperfield** [comment]
> But the behavior with "A" and with { or } is not a known issue. The shortcut for reduce stretch is supposed to be "{". On most keyboards, this is the same as Shift+[. Is that not how it works on your keyboard? Please explain more about the keyboard layout you are using.

Then these are real issues. 
1. <kbd>A</kbd> is still shown as the key to add the note "A"
2. <kbd>{</kbd> and <kbd>}</kbd>are not functioned while <kbd>[</kbd> and <kbd>]</kbd> are work.

**Comment by MarcSabatella** [comment]
Right, and so I’m asking you to provide details of your keyboard layout.  Because these keys work normally on other keyboards.

**Comment by David-C0pperfield** [comment]
@MarcSabatella 
I am using a qwerty keyboard on MacBook Pro (sold in Australia, should be same as the US version, see the similar image below)
![mbp-spacegray-gallery2-202206](https://store.storeimages.cdn-apple.com/8756/as-images.apple.com/is/mbp-spacegray-gallery2-202206?wid=4000&hei=3074&fmt=jpeg&qlt=90&.v=1664495936955)

**Comment by MarcSabatella** [comment]
Weird.  Let's see if anyone else on a Mac can reproduce...

**Comment by MagnusWa** [comment]
I am able to reproduce the issue with the 'A' going back to the beginning of current measure on a mac with swedish keyboard layout.

**Comment by reggoboy** [comment]
I have reported a case with Stretch being mapped to [brackets] rather than {braces} here:

https://musescore.org/en/comment/1160972#comment-1160972

**Comment by syku01** [comment]
The same issue here.

(1) This can be reproduced on both my Mac mini (Apple M1) with macOS Ventura 13.1 (22C65) and my MacBook (Intel) with macOS Monterey 12.6.2.

(2) All the keyboards may reproduce this issue, including keyboard on MacBook, external USB keyboard, Bluetooth keyboard and the software emulated Accessibility Keyboard on the screen.

(3)
The original macOS settings of "Language and Region" on my MacBook was:
Region: Taiwan
Language: Chinese Traditional-Primary, English (US) , Japanese

When I change Region to "US" and Language to "English (US)-Primary" and try Help > Reverting to Factory. The issue disappears. Language setting in Musescore 4 becomes "system default".

Then I change back my macOS setting to Taiwan and Chinese Traditional-Primary. Somehow Musescore 4 shows everything in "Simplified Chinese (China)" instead of "Traditional Chinese (Taiwan)". Since Simplified Chinese is very different from Traditional Chinese (and Taiwan is totally different from China), I have to switch the language setting in Musescore 4 from "system default" to "Traditional Chinese (Taiwan)", which is the last item in the language list. Then the same issue appears again. The cursor jumps back to the beginning of the bar whenever I press A.

I feel that the issue is highly related to the language setting in Musescore 4.

**Comment by floriskruisselbrink** [comment]
Same issue here. 
Macbook Pro 16” 2019 and Macbook Pro 15” 2017.
Keyboard layout or hardware doesn’t make a difference (internal keyboard Dutch layout, usb Apple keyboard, usb Dell keyboard). It is impossible to enter the note A. 
I haven’t tried to fiddle with the language settings, but on my old macbook Dutch is the default language (and Musescore 4 uses it as well), on my new macbook English (US) is the default language (and Musescore 4 uses it). Both exhibit the same problem.

**Comment by Chevy58** [comment]
Not working for me unfortunately.

**Comment by Fince10086** [comment]
> by assigning some key to the "Go to Top Staff" function

It works! many thanks for you @syku01

**Comment by MarcSabatella** [comment]
The problem is quite simple - there is a spurious empty entry for the top-staff command in the shortcut definition files.  Somehow, on certain keyboards, pressing certain keys sends an extra keycode that triggers this command.

I have submitted a PR that removes this entry from the default shortcut file.  THis should fix the problem, but I have no way of testing it - maybe someone with a Mac can check it out by using one of the known problematic layouts (Swedish, Dutch, Taiwanese?)

**Comment by zacjansheski** [comment]
We aren't able to reproduce/test this on our end.  Could anyone confirm if Marc's PR resolves this issue?

https://github.com/musescore/MuseScore/pull/19122

**Comment by MarcSabatella** [comment]
That’s because it’s not fixed until 4.2.  Try the beta (see the Announcements forum on musescore.org).  Or just use the workaround for now - set a custom shortcut for “go to top staff”, in Edit / Preferences / Shortcuts.

**Comment by Benjamin-Benji** [comment]
The solution does not seem to work. However maybe it is close.
However when pressing A it does not jump to top staff. Instead it toggles the focus around. See attached video.

Uploading Skærmoptagelse 2024-08-31 kl. 19.27.34.mov…

**Comment by Benjamin-Benji** [comment]
The solution does not seem to work. However maybe it is close.
When pressing A it does not jump to top staff. Instead it toggles the focus around. See attached Gif
![Pressing A in musescore  2024-09-01 kl  12 08 35](https://github.com/user-attachments/assets/0889c111-1174-4205-b2de-0a58ac3560c5)

**Comment by cbjeukendrup** [comment]
See #20866 and the Pull Request linked to that issue

**Comment by Benjamin-Benji** [comment]
> #20866

Thank you for the link and reply regarding gif/.mov.
